app/uk/gov/hmrc/agentclientrelationships/services/CreateRelationshipsService.scala

```diff
@@ -84,13 +80,13 @@ class CreateRelationshipsService @Inject()(
                 else Future.successful(DbUpdateFailed)
             }
         }
-
         for {
+          agentUser            <- retrieveAgentUser(arn)
```

Comment:
we retrieve the agent details in advance of any calls so nothing further is done if they are not available